File: python/js vs python/estruturas.py
# array
vegetais = ["Repolho", "Nabo", "Rabanete", "Cenoura"]

# acesso a index
vegetais[0]

# adiciona um elemento ao array
vegetais.append("Couve")
# podemos utilizar a função insert especificando uma posição

# removendo um elemento
vegetais.remove("Nabo")

# removendo o ultimo elemento
vegetais.pop()

# buscando um elemento
vegetais.index('Repolho')

# verificando o tamanho de um array
len(vegetais)

# dicionário (equivalente a objeto em python)
dicionario = {
    "nome": "um objeto",
    "descricao": "de madeira",
    "anos_de_uso": 2,
}

# acessando propriedades (valores atráves de chaves em python)
print(dicionario["nome"])
# não é possível acessar utilizando a notação de ponto: dicionario.nome
dicionario.get("nome", "") # o segundo parâmetro é um valor default

# deletando uma propriedade
del dicionario["descricao"]

# elementos do dicionario
dicionario.items()

# existem outros métodos como .keys() e .values()
# para acesso de chaves e valores em dicionarios

File: python/estruturas.js
// array
var vegetais = ['Repolho', 'Nabo', 'Rabanete', 'Cenoura'];

// acesso a index
vegetais[0];

// adiciona um elemento ao array
vegetais.push('Couve');

// removendo um elemento
const index = vegetais.indexOf('Nabo');
vegetais = vegetais.splice(index, 1);

// removendo o ultimo elemento
vegetais.pop();

// buscando um elemento
vegetais.indexOf('Repolho');

// tamanho de um array
vegetais.length


// objeto
var objeto = {
    nome: "um objeto",
    descricao: "de madeira",
    anosDeUso: 2,
};

// acessando propriedades
console.log(objeto.nome);
console.log(objeto["nome"]);
// undefined se não existir

// deletando uma propriedade
delete objeto.descricao;

// elementos do objeto
Object.entries(objeto)
